### PR TITLE
Fixes for Mosek v10

### PR DESCRIPTION
**Pull request by blegat**
Follow up from https://github.com/jump-dev/MosekTools.jl/pull/105
I went through the failing tests and either provided fixes or excluded the expected failures.

### CODE REVIEW

**Comment by ulfworsoe** [comment]
That certainly helped. There are 4 errors left, all of which are related putting conic constraints on overlapping variable vectors.

I have added support for the affine conic constraints in MOSEK v10 (constraint of the form `Ax+b in C`), so if the constraints were added with `AffineVectorFunction` instead of `VectorOfVariables` it would probably solve this problem. Also, the vector variable in MOSEK may removed from the API within 2 or 3 major versions.

File: test/runtests.jl [hunk 1]
```diff
@@ -27,85 +27,85 @@ function MosekOptimizerWithFallback()
 end
 
 
-@testset "SolverName" begin
```
Comment (odow) [inline test/runtests.jl:30]: Were these changes intentional?

Comment (blegat) [reply test/runtests.jl:30]: Yes, this is WIP, I still have a few tests to exclude or fix
